apps/platform-app/src/app/[locale]/(profile)/_actions/mutations/saveMyProfile.ts

```diff
@@ -125,7 +125,7 @@ export const saveMyProfile = withSentry(async (payload: ProfileModel) => {
 
   const updatedProfile = await updateProfileById(foundProfile.id, updatedData)
 
-  if (updatedProfile.state === PublishingState.PENDING) {
+  if (options?.saveWithPublish && updatedProfile.state === PublishingState.PENDING) {
```

Comment:
**issue:** It will never work, because you hardcoded state to DRAFT, so it will never be PENDING to meet this condition.

```ts
const updatedState = hasCommonFields(changedFields, profilePendingFields)
    ? PublishingState.PENDING
    ? PublishingState.DRAFT
    : payload.state
```

Reply:
This is indeed how I intended o leave this. The task was to make the "save and preview" button in Edit Profile NOT fire `runEvaluateProfileAgent()`. I could delete this whole block, but decided to leave it if we want to add "save and publish" button in the future.